File: src/algorithms/running-sum-of-1d-array.js
/**
 * Given an array nums, returns the running sum of nums.
 * 
 * A running sum is defined as runningSum[i] = sum(nums[0]...nums[i]).
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number[]} - Returns the running sum of an array
 */
function runningSum (nums) {
  for (let i = 1; i < nums.length; i++) {
    nums[i] += nums[i - 1];
  }
  return nums;
}

File: src/algorithms/running-sum-of-1d-array.py
def runningSum(nums):
    """Given an array nums. We define a running sum of an array as
    runningSum[i] = sum(nums[0]…nums[i]). Return the running sum of nums.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        list[int]: Returns the running sum of nums.
    """
    res = []
    total = 0
    for num in nums:
        total += num
        res.append(total)
    return res
